sycl/include/CL/sycl/stream.hpp

```diff
@@ -186,6 +225,19 @@ class stream {
     }
   }
 
+#ifdef __SYCL_DEVICE_ONLY__
+  void __init() {
+    // Calculate work item id inside work group, this should be done once, that
+    // is why this is done in _init method, call to __init method is generated
+    // by frontend. As a result each work item will write to its own flush
+    // buffer.
+    FlushBufferSize = FlushSize[0].load();
+    WIOffsetAcc[0].store(0);
+    detail::workGroupBarrier();
```

Comment:
Is it possible to have `WIOffsetAcc` zero-initialized and get rid of the barrier?

Reply:
I haven't found possibility to do so. Please share with me if you have any ideas

Reply:
So the `WIOffsetAcc` is a local pointer, and each WI increments it by `FlushBufferSize` to determine offset for this WI, right? Why not just compute the offset based on a WI id like this: `WIOffset = get_local_id() * FlushBufferSize`?

Reply:
Yes, correct. I was not able to use this approach WIOffset = get_local_id() * FlushBufferSize because I need instance of sycl::id object to call get_local_id.

Reply:
> I need instance of sycl::id object to call get_local_id.

You can `#include <CL/__spirv/spirv_vars.hpp>` and use `__spirv_BuiltInLocalInvocationId` variables declared there from any function you need. They are lowered to the same SPIR-V as with OpenCL get_local_id().

Reply:
I didn't know that, this makes sense, thx. Then I will prepare follow up improvements/fixes for committed patch.